src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+            return union.stream().map(a -> a.getClass().getSimpleName()).collect(Collectors.toList());
+        }
+
+        public static List<String> getAllAnnotationGroupNames() {
+            return Stream.of(AnnotationGroup.values()).map(v -> v.name()).collect(Collectors.toList());
+        }
+
+        public List<InfoFieldAnnotation> createInfoFieldAnnotations() {
+            final List<InfoFieldAnnotation> all = makeAllInfoFieldAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<InfoFieldAnnotation> makeAllInfoFieldAnnotations() {
+            return Arrays.asList(
+                    new InbreedingCoeff(Standard, ActiveRegionBased),
+                    new Coverage(Standard, ActiveRegionBased),
+                    new ReadPosRankSumTest(Standard, ActiveRegionBased),
+                    new BaseQualityRankSumTest(Standard, ActiveRegionBased),
+                    new MappingQualityRankSumTest(Standard, ActiveRegionBased),
+                    new ChromosomeCounts(Standard, ActiveRegionBased),
+                    new StrandOddsRatio(Standard, ActiveRegionBased),
+                    new FisherStrand(Standard, ActiveRegionBased),
+                    new RMSMappingQuality(Standard, ActiveRegionBased),
+                    new QualByDepth(Standard, ActiveRegionBased),
+                    new MappingQualityZero(ActiveRegionBased),
+                    new SampleList(),
+                    new ClippingRankSumTest(ActiveRegionBased),
+                    new LikelihoodRankSumTest(ActiveRegionBased),
+                    new TandemRepeat(ActiveRegionBased),
+                    new GenotypeSummaries(ActiveRegionBased)
+                    //        new PossibleDeNovo(),  //TODO enable this
+            );
+        }
+
+        public List<GenotypeAnnotation> createGenotypeAnnotations() {
+            final List<GenotypeAnnotation> all = makeAllGenotypeAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<GenotypeAnnotation> makeAllGenotypeAnnotations() {
+            return Arrays.asList(
+                    new BaseQualitySumPerAlleleBySample(Standard, ActiveRegionBased),
+                    new DepthPerAlleleBySample(Standard, ActiveRegionBased),
+                    new OxoGReadCounts(ActiveRegionBased)
+            );
+        }
+
+        private <T extends VariantAnnotation> List<T> filterAnnotations(final List<T> all) {
+            final SortedSet<T> annotations = new TreeSet<>(Comparator.comparing(t -> t.getClass().getSimpleName()));
+
+            for (final T t : all){
+                //if any group matches requested groups, it's in
+                if (t.getGroups().stream().anyMatch(g -> annotationGroupsToUse.contains(g.name()))){
+                    annotations.add(t);
+                }
+                if (annotationsToUse.contains(t.getClass().getSimpleName())){
+                    annotations.add(t);
+                }
```

Comment:
Perhaps it makes more sense for `annotationsToExclude` to be handled here as well, to centralize all the filtering of annotations?